locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -197,24 +203,92 @@ pub trait UnlockWithPenaltyModule:
         token_amount * penalty_percentage / MAX_PERCENTAGE as u64
     }
 
-    fn burn_penalty(&self, token_id: TokenIdentifier, fees_amount: &BigUint) {
+    fn burn_penalty(&self, token_id: TokenIdentifier, token_nonce: Nonce, fees_amount: &BigUint) {
         let fees_burn_percentage = self.fees_burn_percentage().get();
         let burn_amount = fees_amount * fees_burn_percentage as u64 / MAX_PERCENTAGE as u64;
         let remaining_amount = fees_amount - &burn_amount;
 
         if burn_amount > 0 {
-            self.send().esdt_local_burn(&token_id, 0, &burn_amount);
+            self.send()
+                .esdt_local_burn(&token_id, token_nonce, &burn_amount);
         }
         if remaining_amount > 0 {
-            self.send_fees_to_collector(token_id, remaining_amount);
+            if self.fees_from_penalty_unlocking().is_empty() {
+                // First fee deposit of the week
+                self.fees_from_penalty_unlocking()
+                    .set(NonceAmountPair::new(token_nonce, remaining_amount));
+            } else {
+                self.merge_fees_from_penalty(token_nonce, &remaining_amount)
+            }
+        }
+
+        // Send fees to FeeCollector SC
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_week_fee_sent_to_collector = self.last_week_fee_sent_to_collector().get();
+        if current_epoch >= last_week_fee_sent_to_collector + EPOCHS_PER_WEEK {
+            self.send_fees_to_collector();
+            self.last_week_fee_sent_to_collector().set(current_epoch);
         }
     }
 
-    fn send_fees_to_collector(&self, token_id: TokenIdentifier, amount: BigUint) {
+    /// Merges new fees with existing fees and saves in storage
+    fn merge_fees_from_penalty(&self, token_nonce: Nonce, new_fee_amount: &BigUint) {
```

Comment:
Why duplicate the code ? Please use the already existing MERGE function

Reply:
Done.